### ISSUE TITLE
[Gateway-API] Incorrect Route Matching with Nested PathPrefix 

### ISSUE THREAD

**Issue by erismaster** [istio/istio#52714]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

Hi. We are migrating a bunch of services to Gateway-API with Istio as the implementation. One of the service teams is relying on a bunch of nested PathPrefix routes to route different paths to different backends. One of the paths includes a `:` in the path, which appears to break/not correctly match the right httproute rule. Below are snippets of our httproutes that are being affected:

```
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: myapp  
 rules:
  - backendRefs:
    - group: ""
      kind: Service
      name: myapp-ui
      port: 80
      weight: 1
    matches:
    - path:
        type: PathPrefix
        value: /myapp
 ```
 
 This path works and routes properly.
        
  ```  
apiVersion: gateway.networking.k8s.io/v1
kind: HTTPRoute
metadata:
  name: myappapiitems
  rules:
  - backendRefs:
    - group: ""
      kind: Service
      name: myapp-service
      port: 8000
      weight: 1
    matches:
    - path:
        type: PathPrefix
        value: /myapp/api/items
 ```

Requests to `/myapp/api/items` also work, however requests to `/myapp/api/items:publish` do _not_ work. Requests to `/myapp/api/items/dafdsafd` also work. It seems there is something with the non-slash character. If I use another char, like @, I get the same misbehaving route matching. 

If I look at the access logs in istio, I can see that requests with `:publish` in the path end up matching the httproute for `/myapp`
The upstream_cluster ends up being `"upstream_cluster":"outbound|80||myapp-ui.mynamespace.svc.cluster.local"` and the route_name is `"route_name":"my-namespace.myapp.0`.  

If we create a new HTTPRoute with a PathPrefix of `/myapp/api/items:publish`, Istio then routes the traffic to the correct backend (this solution is not feasible for us as we do not know all the specific routes, which is why we're using PathPrefix).

I then tested if this same behavior occurs with the Istio CRDs (Gateway and VirtualService). I recreated the setup and a VirtualService that looks like:

```
apiVersion: networking.istio.io/v1
kind: VirtualService
metadata:
  name: myapp-test
spec:
  hosts:
  - <redacted>
  gateways:
  - my-namespace/my-test
  http:
  - match:
    - uri:
        prefix: /myapp/api/items
    route:
    - destination:
        host: myapp-service
        port:
          number: 8000
  - match:
    - uri:
        prefix: /myapp
    route:
    - destination:
        host: myapp-ui
        port:
          number: 80
```

The virtualservice routes requests to `/myapp` route correctly to the UI pod/backend, and requests to `/myapp/api/items` route to the myapp-service backend, as well as `/myapp/api/items:publish`. 

Tldr; When using Gateway-API HTTPRoute, routes with non-slash character don't correctly match PathPrefix. 

### Version

```Text
Istio control plane: 1.22.2
Gateway API: 1.1 Stable (standard-install) API's
```

**Comment by httpdigest** [comment]
It looks like this works as specified. Please read the Gateway API spec, for example: https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io%2fv1.PathMatchType

It explicitly states that "PathPrefix" matches only on slash-delimited path segments. It is not just a string prefix match.
